src/Tasks/Microsoft.NET.Build.Tasks/LockFileCache.cs

```diff
@@ -50,13 +54,79 @@ private static string GetTaskObjectKey(string lockFilePath)
 
         private LockFile LoadLockFile(string path)
         {
-            if (!File.Exists(path))
+            // LockFileUtilties.GetLockFile has odd error handling:
```

Comment:
Add a link to the NuGet bug you filed.  Consider updating or filing new bugs for the other issues.